lib/reversetunnel/api_with_roles.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reversetunnel
+
+import (
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/services"
+
+	"github.com/gravitational/trace"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// NewTunnelWithRoles returns new authorizing tunnel
+func NewTunnelWithRoles(tunnel Tunnel, roles services.RoleSet, ap auth.AccessPoint) *TunnelWithRoles {
+	return &TunnelWithRoles{
+		tunnel: tunnel,
+		roles:  roles,
+		ap:     ap,
+	}
+}
+
+// TunnelWithRoles authorizes requests
+type TunnelWithRoles struct {
+	tunnel Tunnel
+
+	// roles is a set of roles used to check RBAC permissions.
+	roles services.RoleSet
+
+	ap auth.AccessPoint
+}
+
+// GetSites returns a list of connected remote sites
+func (t *TunnelWithRoles) GetSites() ([]RemoteSite, error) {
+	clusters, err := t.tunnel.GetSites()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	out := make([]RemoteSite, 0, len(clusters))
+	for _, cluster := range clusters {
+		if _, ok := cluster.(*localSite); ok {
+			out = append(out, cluster)
```

Comment:
inject local cluster as first element, to be consistent

Reply:
it's always first in this package as returned by (s *server) in srv.go